### PR TITLE
New module: platform agnostic way to manage local users on network devices (network/net_user.)

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

net_user platform agnostic module [WIP]

##### ISSUE TYPE

 - New Module Pull Request

##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

File: lib/ansible/modules/network/net_user.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  users:
+    description:
+      - The set of username objects to be configured on the remote
+        network device. The list entries can either be the username
+        or a hash of username and properties. This argument is mutually
+        exclusive with the C(name) argument.
+  username:
+    description:
+      - The username to be configured on the remote network device.
+        This argument accepts a stringv value and is mutually exclusive
```
Comment (gundalow) [inline lib/ansible/modules/network/net_user.py]: `stringv`?

File: lib/ansible/modules/network/net_user.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        in the device active configuration
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: create a new user
+  net_user:
+    username: ansible
+    sshkey: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
```
Comment (gundalow) [inline lib/ansible/modules/network/net_user.py]: This is a good example and is demonstrates why the module doesn't need to accept a filename

File: lib/ansible/modules/network/net_user.py (hunk 1 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/net_user.py]: Looks like typo. good catch @gundalow thanks.

**Review by rcarrillocruz — APPROVED** [review]

**Comment by trishnaguha** [comment]
Merging this PR as the all tests for this PR are green.
The failed test seems to be caused by some other module on OSX. https://app.shippable.com/github/ansible/ansible/runs/24528/6/tests

**Pull request merged**